File: ruby/bob/bob.rb

class Bob

  def hey(remark)
    if check_if_yelling?(remark) && check_for_letters?(remark)
      'Whoa, chill out!'
    elsif check_if_question?(remark)
      'Sure.'
    elsif remark.tr(" \t", '') == ''
      'Fine. Be that way!'
    else
      'Whatever.'
    end
  end

  def check_for_letters?(remark)
    valid_letters = ("a".."z").to_a
    remark.downcase.chars.any? do |letter|
      valid_letters.include?(letter)
    end
  end

  def check_if_yelling?(remark)
    remark.chars.all? do |letter|
      letter == letter.upcase
    end
  end

  def check_if_question?(remark)
    remark[-1] == '?'
  end

end

File: javascript/bob/bob.js
//
// This is only a SKELETON file for the "Bob" exercise. It's been provided as a
// convenience to get you started writing code faster.
//

var Bob = function() {};

Bob.prototype.hey = function(input) {

var inputOnlyLetters = input.replace(/[^a-zA-Z]+/g, '');
var onlySpaces = input.replace(/[^a-zA-Z0-9]+/g, '');

function noLetter (input) {
	return inputOnlyLetters === '';
};

function upperCase (input) {
	return input.toUpperCase() === input;
};

function isQuestion (input) {
	return input.slice(-1) === '?'; 
};

if (onlySpaces === '') {
	return 'Fine. Be that way!';
} else if (upperCase(input) && !noLetter(input)) {
	return 'Whoa, chill out!';
} else if (isQuestion(input)) {
	return 'Sure.';
} else {
	return 'Whatever.';
}

};

module.exports = Bob;
